### PR TITLE
feat: support contour terminal

### PR DESCRIPTION
**Pull request by TobisLee**
Contour is a modern C++ TE. See https://github.com/contour-terminal/contour

But there's a tearing(?) problem in the image preview, and I'm not sure if it's a Contour issue or a Yazi issue.
My workaround is to disable the status line:
```yml
# in `.config/contour/contour.yml`
status_line:
    # Either none or indicator.
    display: none
```
图片

### CODE REVIEW

**Comment by sxyazi** [comment]
Contour already supports DA1, does it not work for you? https://github.com/contour-terminal/contour/issues/8

**Comment by TobisLee** [comment]
It works, but I find the these in yazi.log:
```txt
  2024-09-02T09:42:08.752730Z  WARN yazi_adapter::emulator: [Adapter] No special environment variables detected
    at yazi-adapter/src/emulator.rs:73

  2024-09-02T09:42:08.752747Z  WARN yazi_adapter::emulator: [Adapter] Unknown TERM_PROGRAM: 
    at yazi-adapter/src/emulator.rs:87

  2024-09-02T09:42:08.752753Z  WARN yazi_adapter::emulator: [Adapter] Unknown TERM: contour
    at yazi-adapter/src/emulator.rs:95
```

File: yazi-adapter/src/emulator.rs [hunk 1]
```diff
@@ -67,6 +69,7 @@ impl Emulator {
 			("WT_Session", Self::Microsoft),
 			("VSCODE_INJECTION", Self::VSCode),
 			("TABBY_CONFIG_DIRECTORY", Self::Tabby),
+			("TERMINAL_NAME", Self::Contour),
```
Comment (sxyazi) [inline yazi-adapter/src/emulator.rs:72]: This can't be used as a "special" env variable as it's not "special"

Comment (TobisLee) [reply yazi-adapter/src/emulator.rs]: I deleted it.

**Pull request closed without merging**